### PR TITLE
{CI} Use `azdev` and `pytest` for full test

### PR DESCRIPTION
**Pull request by jiasli**
## Description

As `nosttests` has actually been deprecated (the repo https://github.com/nose-devs/nose hasn't been updated since 2016), the full test should be moved to `azdev` and `pytest`.

## Known issues

- Coverage tools are installed by the old test pipeline, but is never used:
  https://github.com/Azure/azure-cli/blob/e7186c8160568291edb67169e379343c6312ab7a/scripts/ci/test_automation.sh#L23-L24
- Due to the lack of process isolation (https://github.com/pytest-dev/pytest/issues/6372), `azdev` calls `pytest` to run tests sequentially. This causes the pipeline running time to double compared to that of `nosttests`.

### CODE REVIEW

**Comment by yonzhan** [comment]
CI

**Review by fengzhou-msft — APPROVED** [review]

File: azure-pipelines-full-tests.yml [hunk 1]
```diff
@@ -13,144 +13,74 @@ pr:
       - '*'
 
 jobs:
-  - job: ExtractMetadata
-    displayName: Extract Metadata
+- job: AutomationTest
+  displayName: Automation Test (Profile Latest)
+  timeoutInMinutes: 120
+  pool:
+    vmImage: 'ubuntu-16.04'
```
Comment (qwordy) [inline azure-pipelines-full-tests.yml:20]: Is 16.04 too old? Consider 18.04 or 20.04

File: azure-pipelines-full-tests.yml [hunk 2]
```diff
@@ -13,144 +13,74 @@ pr:
...
+  displayName: Automation Test (Profile 2020-09-01)
+  timeoutInMinutes: 120
+  pool:
+    vmImage: 'ubuntu-16.04'
+  strategy:
+    matrix:
+      Python36:
+        python.version: '3.6'
+      Python38:
+        python.version: '3.8'
```
Comment (qwordy) [inline azure-pipelines-full-tests.yml:44]: Where is 3.7?

File: .azure-pipelines/templates/automation_test.yml [hunk 3]
```diff
@@ -1,6 +1,16 @@
+# Azure DevOps task for automation test using azdev and pytest
+
 parameters:
-  pythonVersion: ''
-  profile: ''
+- name: pythonVersion
+  type: string
+  default: ''
```
Comment (qwordy) [inline .azure-pipelines/templates/automation_test.yml:6]: What does '' mean? How about set default value 3.8?

File: azure-pipelines-full-tests.yml (hunk 1 above)
Comment (jiasli) [reply azure-pipelines-full-tests.yml:20]: Will update in https://github.com/Azure/azure-cli/pull/17368 to keep in sync with main pipeline.

File: azure-pipelines-full-tests.yml (hunk 2 above)
Comment (jiasli) [reply azure-pipelines-full-tests.yml:44]: We package with Python 3.6 and currently we support up to 3.8. @fengzhou-msft

File: .azure-pipelines/templates/automation_test.yml (hunk 3 above)
Comment (jiasli) [reply .azure-pipelines/templates/automation_test.yml:6]: `''` means empty string. I copied it as-is. It also appears here:

**Review by qwordy — APPROVED** [review]

**Pull request merged**